File: Data Structures & Algorithms/subarray-sum-equals-k/submission-2.py
from collections import Counter
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        hsh = Counter()
        hsh[0] = 1
        res = curr_sum = 0
        
        for num in nums:
            curr_sum += num
            diff = curr_sum - k

            res += hsh[diff]
            hsh[curr_sum] = 1 + hsh[curr_sum]
        return res

File: Data Structures & Algorithms/majority-element/submission-2.go
func majorityElement(nums []int) int {
    counter := 1
    curr := nums[0]
    for i := 1; i < len(nums); i++ {
        num := nums[i]
        if num != curr {
            counter--
            if counter == 0 {
                counter = 1
                curr = nums[i]
            }
        } else {
            counter++
        }
    }
    return curr
}
